packages/loader/container-loader/src/container.ts

```diff
@@ -151,6 +151,34 @@ export enum ConnectionState {
     Connected,
 }
 
+// It specifies the summary taken by the serialize api on detached container.
+export interface IDetachedContainerSnapshot {
+    summary: ISummaryTree | ISnapshotTree,
+    // In format version >= 0.1, we would always have summary. However due to back-compat
+    // we would always have ISnapshotTree for version 0.0.
+    formatVersion: string,
+}
+
+// This function converts the snapshot taken in detached container(by serialize api) to snapshotTree with which
+// a detached container can be rehydrated.
+export const convertDetachedContainerSnapshot = (detachedContainerSnapshot: IDetachedContainerSnapshot) => {
+    let snapshotTree: ISnapshotTree;
+    if (detachedContainerSnapshot.formatVersion === "0.1") {
+        const summaryTree = detachedContainerSnapshot.summary as ISummaryTree;
+        const protocolSummaryTree = summaryTree.tree[".protocol"] as ISummaryTree;
+        const appSummaryTree = summaryTree.tree[".app"] as ISummaryTree;
+        assert(protocolSummaryTree !== undefined && appSummaryTree !== undefined,
+            "Protocol and App summary trees should be present");
+        snapshotTree = convertProtocolAndAppSummaryToSnapshotTree(
+            protocolSummaryTree,
+            appSummaryTree,
+        );
+    } else {
+        snapshotTree = detachedContainerSnapshot.summary as ISnapshotTree;
```

Comment:
There is no usage in old format, so you can likely remove it / replace to assert